core/drivers/imx_uart.c

```diff
@@ -110,11 +110,11 @@ static void imx_uart_putc(struct serial_chip *chip, int ch)
 {
 	vaddr_t base = chip_to_base(chip);
 
-	write32(ch, base + UTXD);
-
-	/* Wait until sent */
-	while (!(read32(base + UTS) & UTS_TXEMPTY))
+	/* Wait until there's space in the TX FIFO */
+	while (read32(base + UTS) & UTS_TXFULL)
 		;
+
+	write32(ch, base + UTXD);
```

Comment:
@jordanrh1 What error do you see with the original logic, write and wait until send?

Reply:
No error but it can be considered an 'improvement'. The code should not assume the FIFO has space initially. It is flush's job to wait for all characters to clear. This seemed to be more in line with what the other serial drivers were doing.